Meta/CMake/lagom_compile_options.cmake

```diff
@@ -46,6 +46,8 @@ if (NOT ENABLE_FUZZERS)
     add_linker_flag_if_supported(LINKER:-Bsymbolic-non-weak-functions)
 endif()
 
+add_link_options(LINKER:--no-undefined)
```

Comment:
This is only a thing on ELF linkers (macOS has sane default semantics: `-undefined error`; no clue about Windows). 

Still for ELF: do we also want `--allow-shlib-undefined`?